src/_data/Auth.js

```diff
@@ -30,13 +30,47 @@ export const signout = (nextState, replace) => {
     store.dispatch(updateToken(''));
     replace({ pathname: '/signin', state: nextState });
 };
+const parseOAuthResponse = (responseUrl) => {
+    const matcher = /acct\d=(\w+)&token\d=([\w-]+)/g;
+    const urlParts = responseUrl.split('?');
+    if (urlParts.length !== 2) {
+        throw new Error('Not a valid url');
+    }
+
+    const params = urlParts[1].split(matcher);
+
+    const accounts = [];
+
+    for (let i = 1; i < params.length; i += 3) {
+        accounts.push({
+            account: params[i],
+            token: params[i + 1],
+        });
+    }
 
+    return accounts;
+};
+const parseUrl = (url) => {
+    const accounts = parseOAuthResponse(url);
+    window.BinaryBoot.accounts = accounts;
+    try {
+        window.localStorage.setItem('boot', JSON.stringify(window.BinaryBoot));
+        window.localStorage.setItem('account', JSON.stringify({ token: accounts[0].token }));
+    } catch (e) {
+        window.console.log('Error while saving boot config', e);
+    }
+};
 export const requireAuthOnEnter = (nextState, replace, callback) => {
     const authorized = store.getState().appState.get('authorized');
     if (!authorized) {
         const oAuthUrl = `https://oauth.binary.com/oauth2/authorize?app_id=${window.BinaryBoot.appId}`;
-        window.open(oAuthUrl);
-        // window.location = oAuthUrl;
+        const winAuth = window.open(oAuthUrl, '_blank', 'location=no');
+        winAuth.addEventListener('loadstop', (e) => {
+             if (e.url.indexOf('acct1') > -1) {
+                 parseUrl(e.url);
+                 winAuth.close();
```

Comment:
A reload should take place here before `winAuth.close();` its a challenge at the moment because our device is not running on a server. Would find a way to by tomorrow.